R/DBI-pool.R

```diff
@@ -47,8 +48,39 @@ setMethod("onDestroy", "DBIConnection", function(object) {
 #' @export
 #' @rdname object
 setMethod("onValidate", "DBIConnection", function(object) {
-  check <- dbGetQuery(object, "SELECT 1")
-  df <- data.frame(1)
-  names(df) <- "1"
-  (check == df)[1,]
+  pool <- attr(object, "..metadata", exact = TRUE)$..pool
```

Comment:
If anything, it should be `object_`, right? Because it is the *object's* metadata, not the pool's. I understand what you mean: it's the object's metadata *that is relevant for pool*, but I think it may still be a little misleading... 
You didn't actually suggest the `..` It wasn't very well thought out. I just wanted to make it privatish-looking. As for the members of `..metadata`, I started using the `..` because I wanted to differentiate between the actual environment variables that have reference semantics (the `..` ones) and our local copies (ex: `state <- ..metadata$..state`). But I don't think that actually makes any difference since we always need the `..metadata$`. So, I'll get rid of the members' `..`s.